pkg/pool-weighted/test/LiquidityBootstrappingPool.test.ts

```diff
@@ -17,275 +17,319 @@ describe('LiquidityBootstrappingPool', function () {
-        let now: BigNumber, startTime: BigNumber, endTime: BigNumber;
-        const endWeights = [fp(0.15), fp(0.25), fp(0.55), fp(0.05)];
-
-        sharedBeforeEach('updateWeightsGradually (start time in the past)', async () => {
-          now = await currentTimestamp();
-          // Start an hour in the past
-          startTime = now.sub(MINUTE * 60);
-          endTime = now.add(UPDATE_DURATION);
-        });
-
-        it('fast-forwards start time if in the past', async () => {
-          await pool.updateWeightsGradually(owner, startTime, endTime, endWeights);
-          const updateParams = await pool.getGradualWeightUpdateParams();
-
-          // Start time should be fast-forwarded to now
-          expect(updateParams.startTime).to.equalWithError(now, 0.001);
+      describe('permissioned actions', () => {
+        context('when the sender is the owner', () => {
+          sharedBeforeEach('set sender to owner', async () => {
+            sender = owner;
+            await pool.init({ from: owner, initialBalances });
+          });
+
+          it('swaps can be enabled and disabled', async () => {
+            await pool.setSwapEnabled(sender, false);
+            expect(await pool.instance.getSwapEnabled()).to.be.false;
+    
+            await pool.setSwapEnabled(sender, true);
+            expect(await pool.instance.getSwapEnabled()).to.be.true;
+          });
+
+          it('disabling swaps emits an event', async () => {
+            const receipt = await pool.setSwapEnabled(sender, false);
+    
+            expectEvent.inReceipt(await receipt.wait(), 'SwapEnabledSet', {
+              swapEnabled: false,
+            });
+          });
+    
+          it('enabling swaps emits an event', async () => {
+            const receipt = await pool.setSwapEnabled(sender, true);
+    
+            expectEvent.inReceipt(await receipt.wait(), 'SwapEnabledSet', {
+              swapEnabled: true,
+            });
+          });
+          
+          it('owner can join and receive BPT, then exit', async () => {
+            const bptBeforeJoin = await pool.balanceOf(owner.address);
+            await expect(pool.joinGivenIn({ from: owner, amountsIn: initialBalances })).to.not.be.reverted;
+    
+            const bptAfterJoin = await pool.balanceOf(owner.address);
+            expect(bptAfterJoin).to.gt(bptBeforeJoin);
+
+            await expect(pool.exitGivenOut({ from: owner, amountsOut: initialBalances })).to.not.be.reverted;
+            const bptAfterExit = await pool.balanceOf(owner.address);
+            expect(bptAfterExit).to.lt(bptAfterJoin);
+          });
+
+          describe('update weights gradually', () => {
+            const UPDATE_DURATION = MINUTE * 60;
+      
+            sharedBeforeEach('grant permission', async () => {
+              const action = await actionId(pool.instance, 'updateWeightsGradually');
+              await pool.vault.grantRole(action, owner);
+            });
```

Comment:
I think we don't need this here, as this is `onlyOwner`. Also, given that the entire only owner / delegated owner flow is so common in all pools, I'd try to eventually find a more concise way of testing that independently - let's not worry about it here.